### PR TITLE
Splitting gRPC service class codegen into its own set of files.

### PR DESCRIPTION
**Pull request by nicolasnoble**

### CODE REVIEW

**Comment by ctiller** [comment]
LGTM

File: src/compiler/cpp_generator.cc [hunk 1]
```diff
@@ -378,6 +421,78 @@ grpc::string GetHeaderServices(const grpc::protobuf::FileDescriptor *file,
...
+
+  vars["filename"] = file->name();
+  vars["filename_identifier"] = FilenameIdentifier(file->name());
+
+  std::vector<grpc::string> parts =
+    grpc_generator::tokenize(file->package(), ".");
+
+  for (auto part = parts.rbegin(); part != parts.rend(); part++) {
+    vars["part"] = *part;
+    printer.Print(vars, "}  // namespace $part$ \n");
```
Comment (yang-g) [inline src/compiler/cpp_generator.cc]: nit: remove the trailing space

**Comment by yang-g** [comment]
LGTM except the nit

Is it recommended that the user should only include the .grpc.pb.h?
The examples in the grpc-common repo should be updated to this also?

File: src/compiler/cpp_generator.cc (hunk 1 above)
Comment (nicolasnoble) [inline src/compiler/cpp_generator.cc]: Done.

**Comment by nicolasnoble** [comment]
Fixed the extra space.

I don't really have any good suggestion for the end user, really. Including .grpc.pb.h will definitely pull the normal .pb.h file as well, so technically they don't need to.

grpc-common will need to be updated as well, yes.

**Pull request merged**